gridt/src/app/core/settings.service.spec.ts

```diff
@@ -93,32 +114,163 @@ describe("SettingsService when authentication fails", () => {
 
 describe("SettingsService when authentication is succesful", () => {
   beforeEach( () => {
+    apiStub = jasmine.createSpyObj('api', {'userIdentity$': mock_id[0]});
+
     secStoreStub = jasmine.createSpyObj(
       'secStore', {'get$': of(mock_id[0]), 'set$': of(true), 'clear$': of(true), 'remove$': of(true)}
     );
 
+    httpClientStub = jasmine.createSpyObj(
+      'httpClient', ['get', 'post', 'put']
+    );
+
     TestBed.configureTestingModule({
       providers: [
         SettingsService,
         {provide: AuthService, useClass: authServiceStub_succes},
+        {provide: ApiService, useValue: apiStub},
         {provide: SecureStorageService, useValue: secStoreStub},
+        {provide: HttpClient, useValue: httpClientStub}
       ]
     });
 
     service = TestBed.get(SettingsService as Type<SettingsService>);
-  })
+    api = TestBed.get(ApiService as Type<ApiService>);
+
+    let tick: (milliseconds: number) => void;
+    let fakeNow = 0;
+    tick = milliseconds => {
+      fakeNow += milliseconds;
+      tick(milliseconds);
+    };
+    asyncScheduler.now = () => fakeNow;
+  });
 
   it("should be created", () => {
     expect(service).toBeTruthy();
   });
 
+  it("should return the local id when server id is unavailable", () => {
+    // When there is no internet, the identity must remain available.
+    apiStub.userIdentity$.and.returnValue( throwError("server unavailable") );
+    service.updateIdentity();
+    expect(service.userIdentity$).toBeObservable(cold('a',{a:mock_id[0]}));
+  });
+
+  it("should overwrite local id when server id differs", () => {
+    // Make sure any updates of the ID are reflected in the local storage.
+    apiStub.userIdentity$.and.returnValue( of(mock_id[1]) );
+    service.updateIdentity();
+    expect(secStoreStub.set$).toHaveBeenCalledWith("identity", mock_id[1]);
+  });
+
+  it("should not overwrite local id when local and server id are equal", () => {
+    // Reduces number of calls
+    apiStub.userIdentity$.and.returnValue( of(mock_id[0]) );
+    service.updateIdentity();
+    expect(secStoreStub.set$).not.toHaveBeenCalled();
+  });
+
+  it("should set local id with server id when local id is unavailable", () => {
+    // This happens for new users, because their localstorage has not been set yet.
+    secStoreStub.get$.and.returnValue(throwError("Oops, localstore not set yet"));
+    apiStub.userIdentity$.and.returnValue( of(mock_id[3]));
```

Comment:
We should create a js lint rule for the spacing in arguments, it's silly that this should be caught by humans.

Reply:
Also I notice that you/we are not consistent with this spacing. It would be good to come to a decision and run a formatter to make the entire repository adopt the rule.